### DISCUSSION TITLE
Plans for user interface updates?

### DISCUSSION THREAD

**Discussion by richb-hanover** [mtxr/vscode-sqltools#1241] [Ideas]
First off, let me say that SQLTools is a _tour de force._ All aspects of it - installation, documentation, wide variety of drivers, operation within the VS Code ecosystem, etc. - are really hard software engineering. Congratulate yourselves on your success.

I am wondering about your plans for the user interface, and how it might help people (like me) doing analysis on data stored in SQL databases, and whether those capabilities are on your roadmap. Is it worth my pulling my questions together and asking here? Many thanks.

**Comment by richb-hanover** [comment]
@gjsjohnmurray As I reflect on my hopes for the roadmap, and notice what might be a similar request in #1239, I guess I will state my biggest request. I currently use Database Browser for SQLite (DB4S) to view tables of my database and the results from queries. The window looks like this:

<img width="1315" alt="image" src="https://github.com/mtxr/vscode-sqltools/assets/1094930/75909693-7ca1-4ecb-a9ce-811faf611e62">

The attributes that I like:

- It's wide: it shows as many columns as possible on my laptop screen
- It puts the tables in a dropdown, so that I don't lose much horizontal screen space
- The downside (of using DB4S) is that I have to copy my SQL queries (that I save in VS Code) and paste them into DB4S. I think it would be slick if I could stay in VS Code...

Is anything like this on your roadmap? Thanks again.

**Comment by gjsjohnmurray** [comment]
@richb-hanover thanks for your appreciation and ideas. We're not currently resourced to undertake this level of development, but are open to development sponsorship from the community, and also willing to consider code contributions.

**Comment by richb-hanover** [comment]
Thanks. I plant these seeds, sometimes they germinate into something wonderful.
